pkg/controller/pod.go

```diff
@@ -334,13 +344,35 @@ func (pnc *PodNetworksController) handleResult(
 			updatedStatus,
 		); setNetworkStatusError != nil {
 			klog.Errorf("error updating pod %s network status: %v", namespacedPodNameString, setNetworkStatusError)
+			shouldDeleteattachmentsAdded = true
+		}
+	}
+
+	var attachmentsToCancel []nadv1.NetworkSelectionElement
+	if shouldDeleteattachmentsAdded {
+		attachmentsToCancel = attachmentsToAdd
+	} else if addfailedIndex != -1 {
+		attachmentsToCancel = append(attachmentsToCancel, attachmentsToAdd[addfailedIndex])
```

Comment:
this is a good point ... 

I agree with @LionelJouin; shouldn't you delete all the attachments you've just added ? i.e. setting the pod back to its original state ? 

otherwise, the pod's network-status and the pod's actual interfaces will not be consistent.